pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -2,38 +2,47 @@ package component
 
 import (
 	"fmt"
+	"reflect"
 
+	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
 	"github.com/golang/glog"
 	"github.com/pkg/errors"
 
+	"github.com/openshift/odo/pkg/devfile/adapters/common"
 	adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
 	"github.com/openshift/odo/pkg/devfile/adapters/docker/utils"
 	versionsCommon "github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/exec"
+	"github.com/openshift/odo/pkg/lclient"
 	"github.com/openshift/odo/pkg/log"
 )
 
+const (
+	envCheProjectsRoot         = "CHE_PROJECTS_ROOT"
+	envOdoCommandRunWorkingDir = "ODO_COMMAND_RUN_WORKING_DIR"
+	envOdoCommandRun           = "ODO_COMMAND_RUN"
+)
+
 func (a Adapter) createComponent() (err error) {
 	componentName := a.ComponentName
 
-	volumeLabels := utils.GetProjectVolumeLabels(componentName)
-
 	supportedComponents := adaptersCommon.GetSupportedComponents(a.Devfile.Data)
 	if len(supportedComponents) == 0 {
 		return fmt.Errorf("No valid components found in the devfile")
 	}
 
 	// Create a docker volume to store the project source code
-	volume, err := a.Client.CreateVolume(volumeLabels)
+	volumeLabels := utils.GetProjectVolumeLabels(componentName)
```

Comment:
Heads up, when my changes go in for the volume changes, you'll need to rebase as I changed this line to try and look up the project source volume first, before blindly creating one.